consensus/src/experimental/buffer_item.rs

```diff
@@ -172,4 +198,46 @@ impl BufferItem {
             Self::Aggregated(aggregated) => aggregated.aggregated_proof.ledger_info().commit_info(),
         }
     }
+
+    pub fn add_signature_if_matched(
+        &mut self,
+        target_commit_info: &BlockInfo,
+        author: Author,
+        sig: Ed25519Signature,
+    ) -> bool {
```

Comment:
it'd be better to be a Result that Error explains why adding signature failed.

Reply:
good point